pkg/operation/botanist/health_check.go

```diff
@@ -470,13 +471,13 @@ func (b *HealthChecker) CheckMonitoringSystemComponents(
 
 // CheckMonitoringControlPlane checks whether the monitoring in the given listers are complete and healthy.
 func (b *HealthChecker) CheckMonitoringControlPlane(
-	namespace string,
+	shoot *shoot.Shoot,
```

Comment:
Don't use `operation` objects here - rather use the namespace and hand in the boolean flag whether this thing wants the alertmanager or not.